src/main/java/de/qabel/desktop/ui/actionlog/ActionlogController.java

```diff
@@ -21,75 +34,175 @@
+		List<DropMessage> dropMessages = getDropMassages(siceDate);
 
-		injectionContext = new HashMap<>();
+		if (dropMessages == null) {
+			return;
+		}
 
-		OtherTextWrapper wrapper = new OtherTextWrapper();
-		wrapper.setText(text);
-		Identity i = clientConfiguration.getSelectedIdentity();
-		Contact contact = new Contact(i, i.getAlias(), i.getDropUrls(), i.getEcPublicKey());
-		wrapper.setContact(contact);
-		injectionContext.put("wrapper", wrapper);
+		for (DropMessage d : dropMessages) {
+			addMessageToActionlog(d);
+		}
+	}
+
+	void addMessageToActionlog(DropMessage dropMessage) {
+		lastDate = dropMessage.getCreationDate();
+		Map<String, Object> injectionContext = new HashMap<>();
+		Contact sender = findSender(dropMessage);
+		injectionContext.put("dropMessage", dropMessage);
+		injectionContext.put("contact", sender);
 		OtherActionlogItemView otherItemView = new OtherActionlogItemView(injectionContext::get);
 		messages.getChildren().add(otherItemView.getView());
 		messageView.add(otherItemView);
-		text = "short Text";
-		injectionContext = new HashMap<>();
-		injectionContext.put("text", text);
-		myItemView = new MyActionlogItemView(injectionContext::get);
+	}
+
+	void addOwnMessageToActionlog(DropMessage dropMessage) {
+		Map<String, Object> injectionContext = new HashMap<>();
+		injectionContext.put("dropMessage", dropMessage);
+		MyActionlogItemView myItemView = new MyActionlogItemView(injectionContext::get);
 		messages.getChildren().add(myItemView.getView());
 		messageView.add(myItemView);
+		textarea.setText("");
+	}
 
-		wrapper = new OtherTextWrapper();
-		wrapper.setText(text);
-		i = clientConfiguration.getSelectedIdentity();
-		contact = new Contact(i, i.getAlias(), i.getDropUrls(), i.getEcPublicKey());
-		wrapper.setContact(contact);
-		injectionContext.put("wrapper", wrapper);
-		otherItemView = new OtherActionlogItemView(injectionContext::get);
-		messages.getChildren().add(otherItemView.getView());
-		messageView.add(otherItemView);
+	private Contact findSender(DropMessage dropMessage) {
 
+		List<Contact> contactList = null;
+
+		try {
+			contactList = contactRepository.findAllContactFromOneIdentity(identity);
+		} catch (EntityNotFoundExcepion entityNotFoundExcepion) {
+			entityNotFoundExcepion.printStackTrace();
+		}
+
+		for (Contact c : contactList) {
+			if (dropMessage.getSenderKeyId().equals(c.getEcPublicKey().getReadableKeyIdentifier())) {
+				return c;
+			}
+		}
+
+	return null;
 	}
 
+	DropMessage sendDropMessage(final Contact c, String text) throws QblDropPayloadSizeException {
+		DropMessage d = new DropMessage(identity, text, "dropMessage");
+		final BinaryDropMessageV0 binaryMessage = new BinaryDropMessageV0(d);
+		final byte[] messageByteArray = binaryMessage.assembleMessageFor(c);
+		DropURL dropURL = convertCollectionIntoDropUrl(c.getDropUrls());
 
+		HTTPResult<?> dropResult = dHTTP.send(dropURL.getUri(), messageByteArray);
+
+		if (dropResult.getResponseCode() != 200) {
+			return null;
```

Comment:
throw FailedToSendDropMessageException (or similar)

Btw. 201, 202 and 204 sound ok too.
I think it would be best to wrap the DropHTTP with something that hides this HTTP related code so that the user (the Controller) doesn't have to care about such details. Pretty sure @audax even mentioned there should be a wrapper.